File: openAi/node/testOpenAIAssistant.js
import OpenAI from "openai";
import readline from "readline";
import "dotenv/config.js";

const assistantId = 'asst_bhRd5zAq1b4MgY7oUIlqYucm'; // Replace with your assistant id
const openai = new OpenAI({
  apiKey: process.env.API_KEY
});

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

async function sendMessage(message) {
  let runResponse = await openai.beta.threads.createAndRun({
    assistant_id: assistantId,
    thread: {
      messages: [
        { role: "user", content: message },
      ],
    },
  });

  do {
     await new Promise((resolve) => setTimeout(resolve, 800));
     runResponse = await openai.beta.threads.runs.retrieve(
      runResponse.thread_id,
      runResponse.id
    );     
  } while (runResponse.status === "queued" || runResponse.status === "in_progress") 

  const threadMessages = await openai.beta.threads.messages.list(
    runResponse.thread_id
  );
  const {data} = threadMessages;
  const {content} = data[0];
  console.log(`Response: ${content[0].text.value}`);
}

function main() {
  rl.question('Enter your message: ', async (message) => {
    await sendMessage(message);
    main(); // call main again to prompt for next message
  });
}

main();

File: openAi/python/testOpenAIAssistant.py
from dotenv import load_dotenv
import os
import openai
import time
import sys

load_dotenv()

# Set your OpenAI API key
openai.api_key = os.getenv('API_KEY')

def create_thread(client): 
    return  client.beta.threads.create()

def add_message_to_thread(client, thread_id, message): 
    client.beta.threads.messages.create(
        thread_id,
        role="user",
        content=message
    )    

def create_run(client, thread_id, assistant_id):
    return client.beta.threads.runs.create(
        thread_id=thread_id,
        assistant_id=assistant_id
    )

def retrieve_run(client, thread_id, run_id):
    return client.beta.threads.runs.retrieve(
                thread_id=thread_id,
                run_id=run_id
            )      

def retrieve_response(client, thread_id):
      thread_messages = client.beta.threads.messages.list(thread_id)
      return thread_messages.data[0].content[0].text.value


def main():
    # Initialize the OpenAI client
    client = openai
    
    assistant_id = "asst_bhRd5zAq1b4MgY7oUIlqYucm"

    thread = create_thread(client)

    message = sys.argv[1]
    add_message_to_thread(client, thread.id, message)

    run = create_run(client, thread.id, assistant_id)

    while True:
        run = retrieve_run(client, thread.id, run.id)        
        status = run.status
        
        if status == 'completed':
            break
        time.sleep(1)  # Adjust polling interval as needed

    print(f"Response: {retrieve_response(client, thread.id)}")
    
if __name__ == "__main__":
    main()

